### PR TITLE
Fix payment related bugs

### PR DESCRIPTION
**Pull request by Iamsamwhoami**

### CODE REVIEW

File: Src/Screens/ProfileScreen/Screens/MyPayment/MyPayment.swift [hunk 1]
```diff
@@ -76,7 +76,7 @@ extension MyPayment: Presentable {
                 viewController.present(directDebitSetup, options: [.autoPop])
             }
 
-            if result.data?.isSwitchingBankAccount() ?? false {
+            if result.data?.directDebitStatus == .pending {
```
Comment (meletios) [inline Src/Screens/ProfileScreen/Screens/MyPayment/MyPayment.swift]: i think here should be needs_setup

Comment (Iamsamwhoami) [reply Src/Screens/ProfileScreen/Screens/MyPayment/MyPayment.swift]: @meletios this is for hiding the "Connect autogiro"-button, which is only to be hidden if it's pending, right?

Comment (meletios) [reply Src/Screens/ProfileScreen/Screens/MyPayment/MyPayment.swift]: We need to hide it, when its pending or connected, eller?

Comment (Iamsamwhoami) [reply Src/Screens/ProfileScreen/Screens/MyPayment/MyPayment.swift]: only pending :D

Comment (meletios) [reply Src/Screens/ProfileScreen/Screens/MyPayment/MyPayment.swift]: yeap correct. I thought we are talking about the banner :laughing:

Comment (Iamsamwhoami) [reply Src/Screens/ProfileScreen/Screens/MyPayment/MyPayment.swift]: hehe no

**Pull request merged**